scripts/src/java/org/oppia/android/scripts/maven/GenerateLicenseTexts.kt

```diff
@@ -0,0 +1,423 @@
+    licenseMap[licenseLink] = "license_name_$index"
+  }
+
+  // Add all string-arrays of license names for each dependency.
+  for (index in mavenDependenciesList.indices) {
+    val dependency = mavenDependenciesList[index]
+    val stringArrayElement = doc.createElement("string-array")
+    stringArrayElement.setAttribute("name", "third_party_dependency_licenses_names_$index")
+    dependency.licenseList.forEach { license ->
+      val licenseLink = if (license.primaryLinkType == PrimaryLinkType.SCRAPE_DIRECTLY ||
+        license.primaryLinkType == PrimaryLinkType.SHOW_LINK_ONLY
+      ) {
+        license.primaryLink
+      } else {
+        license.alternativeLink
+      }
+      val indexOfLicenseName = licenseMap[licenseLink]
+      val stringItemElement = doc.createElement("item")
+      stringItemElement.appendChild(doc.createTextNode("@string/$indexOfLicenseName"))
+      stringArrayElement.appendChild(stringItemElement)
+    }
+    rootResourcesElement.appendChild(stringArrayElement)
+  }
+
+  // Add an array that contains string-arrays of license names for each dependnecy.
+  val arrayElement = doc.createElement("array")
+  arrayElement.setAttribute("name", "third_party_dependency_license_names_array")
+  for (index in mavenDependenciesList.indices) {
+    val itemElement = doc.createElement("item")
+    itemElement.appendChild(
+      doc.createTextNode("@array/third_party_dependency_licenses_names_$index")
+    )
+    arrayElement.appendChild(itemElement)
+  }
+
+  rootResourcesElement.appendChild(arrayElement)
+  doc.appendChild(rootResourcesElement)
+  doc.xmlStandalone = true
+  getTransformer().transform(DOMSource(doc), StreamResult(file))
+}
+
+fun retrieveAllLicensesSet(
+  mavenDependencyList: List<MavenDependency>
+): Set<CopyrightLicense> {
+  val copyrightLicensesSet = mutableSetOf<CopyrightLicense>()
+  mavenDependencyList.forEach { dependency ->
+    val licenseList = dependency.licenseList
+    if (licenseList.isEmpty()) {
+      throw Exception(MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE)
+    }
+    licenseList.forEach { license ->
+      val licenseText: String
+      val licenseLink: String
+      when (license.primaryLinkType) {
+        PrimaryLinkType.SCRAPE_DIRECTLY -> {
+          licenseText = scrapeLicenseText(license.primaryLink)
+          licenseLink = license.primaryLink
+        }
+        PrimaryLinkType.SCRAPE_FROM_LOCAL_COPY -> {
+          licenseText = scrapeLicenseText(license.alternativeLink)
+          licenseLink = license.alternativeLink
+        }
+        PrimaryLinkType.SHOW_LINK_ONLY -> {
+          licenseText = license.primaryLink
+          licenseLink = license.primaryLink
+        }
+        else -> throw Exception(MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE)
+      }
+      copyrightLicensesSet.add(
+        CopyrightLicense
+          .newBuilder()
+          .setLicenseName(license.licenseName)
+          .setLicenseText(licenseText)
+          .setLicenseLink(licenseLink)
+          .build()
```

Comment:
Optional, but a slightly nicer way to construct a proto is:

```suggestion
        CopyrightLicense.newBuilder().apply {
          licenseName = license.licenseName
          this.licenseText = licenseText
          this.licenseLink = licenseLink
        }.build()
```

Reply:
No more needed as we are not using this proto class, but made changes for other proto classes.